example-ideal.txt

```diff
@@ -0,0 +1,42 @@
+start toTopLeft
+
+task toTopLeft:
+	start goingLeft
+
+	goingLeft:
+		if isEmpty(Left)
+			move(Left)
+
+		if isWall(Left)
+			next(goingUp)
+
+	goingUp:
+		if isEmpty(Up)
+			move(Up)
+
+		if isWall(Up) and isEmpty(Down)
+			move(Down)
+			next(zigZag)
+
+
+task zigZag:
+	start goingDown
+
+	goingDown:
+		if isEmpty(Down)
+			move(Down)
+
+		if isWall(Down) and isEmpty(Right)
+			move(Right)
+			next(goingUp)
+
+	goingUp:
+		if isEmpty(Up)
+			move(Up)
+
+		if isWall(Up) and isEmpty(Right)
+			move(Right)
+			next(goingDown)
+	
+
+	
```

Comment:
So, hypothetically, there would be a lot of different ways to write the same program here right? You could effectively mix any amount of rules into one task, or divide into several tasks based on what was more intuitive?